drivers/ieee802154/ieee802154_cc13xx_cc26xx.c

```diff
@@ -394,8 +394,12 @@ static void ieee802154_cc13xx_cc26xx_rx_done(
 			len = drv_data->rx_data[i][0];
 			sdu = drv_data->rx_data[i] + 1;
 			seq = drv_data->rx_data[i][3];
-			corr = drv_data->rx_data[i][len--] & 0x3F;
-			rssi = drv_data->rx_data[i][len--];
+			corr = drv_data->rx_data[i][len] & 0x3F;
+			rssi = drv_data->rx_data[i][len-1];
+
+			if (IS_ENABLED(CONFIG_NET_L2_IEEE802154)) {
+				len -= 2;
```

Comment:
Can you add a comment as to why this is necessary? From what I gather, for in `CONFIG_IEEE802154_RAW_MODE`, it's expected that the packet is prepended with correlation and rssi, is that correct?